raft/src/test/java/org/apache/kafka/raft/RaftEventSimulationTest.java

```diff
@@ -712,11 +712,6 @@ void initializeElection(ElectionState election) {
 
             nodes.values().forEach(state -> {
                 state.store.writeElectionState(election);
-                if (election.hasLeader()) {
```

Comment:
So the reason that this is safe is because we no longer try assigning offset other than the end offset?

Reply:
Oh, I think this was just unnecessary. The purpose of this logic is to initialize a certain election state prior to starting the simulation. But the code already initializes the epoch state on startup, so we didn't need this.